docs/DeveloperGuide.md

```diff
@@ -335,19 +348,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. DonnaFin shows an error message.
 
       Use case resumes at step 2.
 
-*{More to be added}*
+**UC03: Locating a contact by name**
+
+**MSS**
+1. User types the command to find the person in the appropriate format.
+2. DonnaFin displays the contacts that match the keyword inputted.
+
+**Extensions**
+* 1a. The user types the command using the wrong format.
+  * 1a1. DonnaFin shows an error message.\
+         Use Case resumes at step 1.
+* 1b. The keyword does not match any contacts.
+  * 1b1. DonnaFin does not display any contact.\
+         Use Case ends.
+
+**UC04: Editing the details of a contact**
+
+**MSS**
+1. User types the command to edit a contact in the appropriate format.
+2. DonnaFin announces that the contact has been successfully updated and displays the new details.
+
+**Extensions**
+* 1a. The user types the command using the wrong format.
+    * 1a1. DonnaFin shows an error message.\
+      Use Case resumes at step 1.
+* 1b. The given index is invalid.
+    * 1b1. DonnaFin shows an error message.\
+      Use Case resumes at step 1.
+  
```

Comment:
Can include US05: Help and US:06 List. Can refer to the addressbook userguide for the commands we can do and think from the commands there. List should be like wanna see everyone or smth. Then help is user wants to know what command to use or smth. Also can consider the saving part.